src/devices/Nmea0183/RoutePoint.cs

```diff
@@ -0,0 +1,140 @@
+        public RoutePoint(string routeName, int indexInRoute, int totalPointsInRoute, string waypointName, GeographicPosition position,
+            Angle? bearingToNextWaypoint, Length? distanceToNextWaypoint)
+        {
+            RouteName = routeName;
+            IndexInRoute = indexInRoute;
+            TotalPointsInRoute = totalPointsInRoute;
+            WaypointName = waypointName;
+            Position = position;
+            BearingToNextWaypoint = bearingToNextWaypoint;
+            DistanceToNextWaypoint = distanceToNextWaypoint;
+        }
+
+        /// <summary>
+        /// The name of the route
+        /// </summary>
+        public string RouteName
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// The index within the route
+        /// </summary>
+        public int IndexInRoute
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// The total number of points on the route
+        /// </summary>
+        public int TotalPointsInRoute
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// The name of this point
+        /// </summary>
+        public string WaypointName
+        {
+            get;
+        }
+
+        /// <summary>
+        /// The position of this point
+        /// </summary>
+        public GeographicPosition Position
+        {
+            get;
+        }
+
+        /// <summary>
+        /// True bearing from this waypoint to the next
+        /// </summary>
+        public Angle? BearingToNextWaypoint
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The distance to the next waypoint
+        /// </summary>
+        public Length? DistanceToNextWaypoint
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Two points are considered equal if the name and the position are equal. The other properties are NMEA-internals and are
+        /// not directly related to the function of the waypoint for the user
+        /// </summary>
+        public bool Equals(RoutePoint? other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return WaypointName == other.WaypointName && Equals(Position, other.Position);
+        }
+
+        /// <summary>
+        /// Equality comparer
+        /// </summary>
+        public override bool Equals(object? obj)
+        {
+            return ReferenceEquals(this, obj) || (obj is RoutePoint other && Equals(other));
+        }
+
+        /// <summary>
+        /// Standard hash function
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return WaypointName.GetHashCode() ^ Position.GetHashCode();
```

Comment:
please combine hash codes as mentioned in the other examples, here and everywhere